Sources/generate-pipeline/RuleCollector.swift

```diff
@@ -132,7 +136,10 @@ final class RuleCollector {
       /// Ignore it if it doesn't have any; there's no point in putting no-op rules in the pipeline.
       /// Otherwise, return it (we don't need to look at the rest of the inheritances).
       guard !visitedNodes.isEmpty else { return nil }
-      return DetectedRule(typeName: typeName, visitedNodes: visitedNodes, canFormat: canFormat)
+      let isOptIn =
+        (_typeByName("SwiftFormatRules.\(typeName)") as? Rule.Type).map { $0.isOptIn } ?? false
```

Comment:
This doesn't seem like it should work since `generate-pipeline` doesn't depend the on `SwiftFormatRules` module. Are you sure `_typeByName` isn't just returning `nil` here and it's ending up `false` by default, yielding the same effective outcome?

If we add the dependency to `SwiftFormatRules`, I think we should make it a stronger precondition failure if `_typeByName` returns nil because that shouldn't happen for any rules that we discover through our syntax walk, and then we should be safer.

Reply:
Good point. Fixed that.